### PR TITLE
fix(Channel): Allow `Channel#delete` to specify a reason

### LINKED ISSUE #6411: .delete() Method on Channel not accepting reason
**Please describe the problem you are having in as much detail as possible:**
.delete() Method on Channels dosen't accept a reason anymore but the api still accepts `X-Audit-Log-Reason header`.
Discord Ref : https://discord.com/developers/docs/resources/channel#deleteclose-channel

The discord.js doc still says that we can input a reason for GuildChannel deletion but typings don't.
The discord.js doc says that we can't input a reason for Channel deletion neither does typings.

I'm not sure if we should be able to tell a reason for deletion on any channel, the discord docs dont say much. But clearly there is mistakes in the library.

**Include a reproducible code sample here, if possible:**
image

https://github.com/discordjs/discord.js/blob/main/src/structures/Channel.js#L96

```js
new GuildChannel(guild).delete()
```

**Further details:**

- discord.js version: 13.1.0
- Node.js version: 16.6.1
- Operating system: Windows
- Priority this issue should have: Low

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Closes #6411

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: src/structures/Channel.js [hunk 1]
```diff
@@ -86,17 +86,18 @@ class Channel extends Base {
...
+   * @param {string} [reason=undefined] The reason for the channel being deleted.
    * @returns {Promise<Channel>}
    * @example
    * // Delete the channel
    * channel.delete()
    *   .then(console.log)
    *   .catch(console.error);
    */
-  delete() {
+  delete(reason = undefined) {
```
Comment (NMW03) [inline src/structures/Channel.js:97]: Do you need to set default as `undefined`? I mean if there is no `reason`, it would be automatically `undefined`

Comment (suneettipirneni) [reply src/structures/Channel.js]: Yeah... I think my inner typescript dev came out, lol

File: src/structures/Channel.js [hunk 2]
```diff
@@ -86,17 +86,18 @@ class Channel extends Base {
 
   /**
    * Deletes this channel.
+   * @param {string} [reason=undefined] The reason for the channel being deleted.
```
Comment (Syjalo) [inline src/structures/Channel.js:89]:
```suggestion
   * @param {string} [reason] Reason for deleting this channel
```

**Review by Syjalo — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
This has been added with #1490
Actual lines, since the PR is somewhat big: https://github.com/discordjs/discord.js/commit/0baa59b679d8542857926d645a02982204b3b80c#diff-6931a2b8bde677910786bd21afe3bc29740277d6510de785acfebca7e616ad89R343-R356

Under the current commit on our main branch:
https://github.com/discordjs/discord.js/blob/71fdf6fae0457bc5f2900a37cc2d28c587e7e54a/src/structures/GuildChannel.js#L536-L549
It also appears on our documentation just fine, when looking under the appropriate class:
https://discord.js.org/#/docs/main/stable/class/GuildChannel?scrollTo=delete

**Comment by suneettipirneni** [comment]
> This has been added with #1490
> Actual lines, since the PR is somewhat big: [0baa59b#diff-6931a2b8bde677910786bd21afe3bc29740277d6510de785acfebca7e616ad89R343-R356](https://github.com/discordjs/discord.js/commit/0baa59b679d8542857926d645a02982204b3b80c#diff-6931a2b8bde677910786bd21afe3bc29740277d6510de785acfebca7e616ad89R343-R356)
> 
> Under the current commit on our main branch:
> https://github.com/discordjs/discord.js/blob/71fdf6fae0457bc5f2900a37cc2d28c587e7e54a/src/structures/GuildChannel.js#L536-L549
> 
> 
> It also appears on our documentation just fine, when looking under the appropriate class:
> https://discord.js.org/#/docs/main/stable/class/GuildChannel?scrollTo=delete

Wasn't aware, I'll go ahead and close this then.

**Pull request closed without merging**